unit_tests/conftest.py

```diff
@@ -129,16 +146,14 @@ def configure_scylla_node(docker_scylla_args: dict, params):  # noqa: PLR0914
                           extra_docker_opts=extra_docker_opts, docker_network=docker_network)
 
     if ssl:
-        curr_dir = os.getcwd()
-        try:
-            os.chdir(Path(__file__).parent.parent)
-            create_certificate(CLIENT_FACING_CERTFILE, CLIENT_FACING_KEYFILE, cname="scylladb",
-                               ca_cert_file=CA_CERT_FILE, ca_key_file=CA_KEY_FILE,
-                               ip_addresses=[scylla.ip_address], dns_names=[scylla.public_dns_name])
-            create_certificate(CLIENT_CERT_FILE, CLIENT_KEY_FILE, cname="scylladb",
-                               ca_cert_file=CA_CERT_FILE, ca_key_file=CA_KEY_FILE)
-        finally:
-            os.chdir(curr_dir)
+        create_certificate(ssl_dir / TLSAssets.DB_CLIENT_FACING_CERT, ssl_dir / TLSAssets.DB_CLIENT_FACING_KEY,
+                           cname="scylladb", ca_cert_file=CA_CERT_FILE, ca_key_file=CA_KEY_FILE,
+                           ip_addresses=[scylla.internal_ip_address], dns_names=[scylla.public_dns_name])
+        create_certificate(ssl_dir / TLSAssets.CLIENT_CERT, ssl_dir / TLSAssets.CLIENT_KEY,
+                           cname="scylladb", ca_cert_file=CA_CERT_FILE, ca_key_file=CA_KEY_FILE)
+
+        request.node.stash[ssl_dir_key] = ssl_dir
+        scylla.__class__.ssl_conf_dir = property(lambda self: ssl_dir)
```

Comment:
Do we really have no better way of adding it here? AFAIK, this is our object, cannot we pass it through a method or a constructor?

Reply:
I think this is still valid. Its our class, we can add the option to pass ssl_conf_dir in a nicer way